internal/marketplaces/types/types.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2024 Stacklok, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package types contains domain models for marketplaces
+package types
+
+import (
+	"github.com/google/uuid"
+
+	"github.com/stacklok/minder/internal/db"
+)
+
+// ProjectContext contains the information needed to create rule types and
+// profiles in a project.
+// This may be useful in various parts of the codebase outside of marketplaces.
+type ProjectContext struct {
+	// Project ID
+	ID uuid.UUID
+	// Provider which profiles/rule types will be linked to
+	Provider *db.Provider
```

Comment:
Is this strictly needed? We're hoping to decouple providers from rule types and profiles to make them more reusable

Reply:
Right now, because creating and updating profiles/rule types requires a provider, I need to pass it along. It is a relatively simple refactor to get rid of the provider and just convert this to a projectID parameter.